frontend/src/components/regions/UserActions/index.jsx

```diff
@@ -22,17 +23,54 @@ const days = [
   "Saturday",
 ];
 
+const matchEventTypesWithColors = ({ eventTypes, eventTypeColors }) => {
+  const arr = [];
+  eventTypes = Object.keys(eventTypes).map((key) => eventTypes[key]);
+
+  eventTypes.forEach((eventType) => {
+    arr.push({
+      id: eventType.id,
+      eventName: eventType.name,
+      eventColor: eventTypeColors[eventType.name],
+    });
+  });
+  return arr;
+};
+
+function mapData({ eventTypesWithColors, activityLogEntries }) {
```

Comment:
This is actually a VERY useful function that I can see being used in other places and not just here. In an effort to    avoid non-dry code, could we perhaps take this function, make it universal and place it in a utility file of some sort?